app/models/maestrano/connector/rails/concerns/entity.rb

```diff
@@ -477,7 +481,11 @@ def batch_calls(array_with_idmap, proc, connec_entity_name, id_update_only = fal
         log_info = id_update_only ? 'with only ids' : ''
         Maestrano::Connector::Rails::ConnectorLogger.log('info', @organization, "Sending batch request to Connec! #{log_info} for #{self.class.normalize_connec_entity_name(connec_entity_name)}. Batch_request_size: #{batch_request[:ops].size}. Call_number: #{(start / request_per_call) + 1}")
         response = @connec_client.batch(batch_request)
-        Maestrano::Connector::Rails::ConnectorLogger.log('debug', @organization, "Received batch response from Connec! for #{self.class.normalize_connec_entity_name(connec_entity_name)}: #{response}")
+
+        response_hash = JSON.parse(response.body)['results'] ? (JSON.parse(response.body)['results']&.map { |h| h['body'] }) : response
```

Comment:
make var for `(response.body)['results']` for readability.

Reply:
@paulstuartparker Need to guard against `response`/`response.body` being nil?

Reply:
Seems like a good idea